pyuvdata/uvbeam.py

```diff
@@ -978,6 +977,43 @@ def __iadd__(self, other):
         self.__add__(other, inplace=True)
         return self
 
+    def get_beam_area(self,stokes='I'):
+        """
+        Computes the integral of the beam, which has units of steradians
+        """
+        if self.beam_type is not 'power':
+            raise ValueError('beam_type must be power')
+        if self._data_normalization.value != 'peak':
+            print self._data_normalization.value
+            raise ValueError('beam must be peak normalized')
+        npix = self.data_array.shape[-1]
+        if stokes == 'I':
+            # Assume A_I = (B_xx + B_yy)/2
+            beam = np.mean(self.data_array[0,0], axis=-3)
+        else:
+            raise NotImplementedError()
+
+        return np.sum(beam, axis=-1) * 4.*np.pi / npix
```

Comment:
I think this is assuming something about how much of the sky the beam model covers and/or the area of the pixels? The beam is not required to cover the full sky.

Also, if this isn't a HEALPix beam I think you need to sum over the last 2 indices.

Reply:
Changed to only allow healpix.